### PR TITLE
[Perf] Refactor LoRAManager to eliminate stream syncs and redundant computations 

### PR DESCRIPTION
**Pull request by lifuhuang**
This PR improves LoRA inference performance by:
- Eliminating unnecessary CUDA stream synchronizations
- Reducing redundant computations

These changes help us achieve our goal of making the LoRA batch initialization process free of CUDA syncs, as outlined in #6961 .

Benchmark results show that this PR, combined with my previous PR (#6960), reduces TTFT (P50) by **31.4%** and ITL (P50) by **34.0%**.

### Benchmark Result
Benchmark Script
```
python3 -m sglang.launch_server --model-path meta-llama/Llama-3.1-8B-Instruct --disable-radix-cache --lora-paths lora=algoprog/fact-generation-llama-3.1-8b-instruct-lora
python3 -m sglang.bench_serving --backend sglang  --num-prompt 480 --request-rate 8 --lora-name lora
```

| | [Baseline ](https://github.com/sgl-project/sglang/commit/9d5fa68b903d295d2b39201d54905c6801f60f7f)| #6960   (merged) | **This PR** (including #6960 ) |
|--------|--------|--------|--------|
| ITL@P95 | 78.42 ms | 68.24 ms (-13.0%) | **52.51 (-33.0%)**  |
| ITL@P50 | 34.36 ms | 32.85 ms (-4.4%) | **22.68 (-34.0%)**  |
| TTFT@P50 | 91.37 ms | 85.52 ms (-6.5%) | **62.65 (-31.4%)**  |

**Before** (See the huge bubble caused by cudaStreamSynchronize)
image

**After** (All cudaStreamSynchronize are removed)
image

### Comparison between CUDA Graph enabled vs disabled
||Baseline | #6861  (merged) | **This PR** |
|--------|--------|--------|--------|
|CUDA Graph Enabled | image | image| image |
|CUDA Graph Disabled | image | image|image|

## Modifications

(Generated by Copilot..)

### LoRA Batch Processing Enhancements:
* Added a new helper function `transfer_adapter_info` in `lora_manager.py` to handle asynchronous transfer of adapter metadata (weight indices, LoRA ranks, and scalings) from the host to the CUDA device. This reduces synchronization overhead and improves performance.
* Improved the initialization of `seg_lens` and `seg_indptr` for CUDA graphs in `init_cuda_graph_batch_info`. These values are now precomputed and remain constant across batches, reducing redundant computations.

### Memory Management Simplifications:
* Simplified the `get_available_buffer_slot` method in `mem_pool.py` by removing the unnecessary return of evicted LoRA UIDs. Eviction logic is now handled internally, improving code clarity.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by Fridge003 — APPROVED** [review]
LGTM

File: python/sglang/srt/lora/lora_manager.py [hunk 1]
```diff
@@ -81,14 +81,25 @@ def init_cuda_graph_batch_info(self, max_bs_in_cuda_graph: int):
...
-                max_len=0,
+                max_len=1,
                 weight_indices=torch.zeros(
                     self.max_bs_in_cuda_graph, dtype=torch.int32
                 ),
                 lora_ranks=torch.zeros(self.max_loras_per_batch, dtype=torch.int32),
                 scalings=torch.zeros(self.max_loras_per_batch, dtype=torch.float),
             )
 
+            # Initialize seg_lens and seg_indptr for CUDA graph as they remain constant
```
Comment (hebiao064) [inline python/sglang/srt/lora/lora_manager.py:92]: is seg_lens a typo?

Comment (Fridge003) [reply python/sglang/srt/lora/lora_manager.py:92]: It's not a typo. It's the name of a variable below.

**Review by Qiaolin-Yu — APPROVED** [review]
Great work! LGTM
